flink-catalog-examples/README.md

```diff
@@ -0,0 +1,273 @@
+    'schema-registry-uri' = 'http://schemaregistry:9092',   --Schema-Registry service URI
+    'default-database' = 'examples',                        --Flink catalog database, mapped to Pravega scope
+    'serialization.format' = 'Avro'                         --data serialization format
+);
+```
+
+### Show Tables
+The Pravega catalog will translate the schema from schema registry service and use it to create the flink table.
+We can check the tables in Pravega catalog: 
+```sql
+USE CATALOG pravega;
+SHOW TABLES;
+DESCRIBE userBehavior;
+SELECT * FROM userBehavior;
+```
+![sql-describe](images/sql-describe.png)
+![query-all](images/query-all.gif)
+
+### Use case 1: Unique Visitors (UV)
+
+One typical use case in the online services is to calculate the cumulative unique visitors continuously, i.e. every ten minutes in one day. 
+For example, the number of UV at 10:00 represents the total number of UV from 00:00 to 10:00.
+This can be easily and efficiently implemented by `CUMULATE` windowing.  
+You can check more about `CUMULATE` and other Windowing TVFs [here](https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/table/sql/queries/window-tvf/).
+
+In order to enrich the table with the event time and watermark information since Pravega catalog do not support
+watermark, we need to extend the existing catalog table like this using [LIKE clause](https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/table/sql/create/#like):
+```sql
+CREATE TABLE default_catalog.default_database.user_behavior (
+  WATERMARK FOR ts as ts - INTERVAL '5' SECOND
+) LIKE userBehavior;
+```
+
+We can have a cumulating window for 10 min step and 1 day max size and get the windows [00:00, 00:10), [00:00, 00:20),
+..., [00:00, 24:00):
+```sql
+SELECT window_start, window_end, COUNT(DISTINCT user_id) as UV
+FROM Table(
+    CUMULATE(Table default_catalog.default_database.user_behavior, DESCRIPTOR(ts), INTERVAL '10' MINUTES, INTERVAL '1' DAY))
+GROUP BY window_start,window_end;
+```
+![query-with-cumulate](images/query-with-cumulate.gif)
+
+### Use case 2: Extract-Transform-Load(ETL) pipeline
+
+The aforementioned use case shows how we can query the data from Pravega streams. 
+In addition, Pravega can also be the downstream sink to persist an SQL query output to construct an end-to-end ETL pipeline.
+
+In the retail industry, transactions at midnight may have more fraud risks, so a typical use case would be that these transactions 
+need to be filtered and persisted again for further fraud-detection analysis. 
+Here we use the `INSERT INTO ... SELECT` clause to contruct an ETL pipeline to read from the data source and 
+persist all the transactions between 00:00 to 03:00 into another Pravega stream.
+
+The Pravega Catalog will first create a stream with table name in Pravega. 
+Then it will convert flink table schema to schema info which can be read by Schema Registry so that this schema can be registered in
```

Comment:
Please review the comments of my first round review carefully and do not resolve the conversation unless you're sure it is fixed.